src/dsp_tools/commands/xmllib/models/dsp_base_resources.py

```diff
@@ -0,0 +1,269 @@
+
+
+@dataclass
+class RegionResource:
+    res_id: str
+    label: str
+    color: str
+    region_of: str
+    geometry: dict[str, Any]
+    comments: list[str]
+    permissions: str = "res-default"
+
+    def __post_init__(self) -> None:
+        _check_and_warn_strings(self.res_id, self.res_id, "Resource ID")
+        _check_and_warn_strings(self.res_id, self.label, "Label")
+        if fail_msg := find_geometry_problem(self.geometry):
+            msg = f"The geometry of the resource with the ID '{self.res_id}' failed validation.\n" + fail_msg
+            warnings.warn(DspToolsUserWarning(msg))
+
+    def serialise(self) -> etree._Element:
+        res_ele = self._serialise_resource_element()
+        res_ele.append(self._serialise_geometry())
+        res_ele.extend(self._serialise_values())
+        if self.comments:
+            res_ele.append(_serialise_has_comment(self.comments, self.res_id))
+        return res_ele
+
+    def _serialise_resource_element(self) -> etree._Element:
+        attribs = {"label": self.label, "id": self.res_id}
+        if self.permissions:
+            attribs["permissions"] = self.permissions
+        return etree.Element(f"{DASCH_SCHEMA}region", attrib=attribs, nsmap=XML_NAMESPACE_MAP)
+
+    def _serialise_values(self) -> list[etree._Element]:
+        return [
+            ColorValue(value=self.color, prop_name="hasColor", resource_id=self.res_id).serialise(),
+            LinkValue(value=self.region_of, prop_name="isRegionOf", resource_id=self.res_id).serialise(),
+        ]
+
+    def _serialise_geometry(self) -> etree._Element:
+        geo_prop = etree.Element(f"{DASCH_SCHEMA}geometry-prop", name="hasGeometry", nsmap=XML_NAMESPACE_MAP)
+        ele = etree.Element(f"{DASCH_SCHEMA}geometry", nsmap=XML_NAMESPACE_MAP)
+        ele.text = str(self.geometry)
+        geo_prop.append(ele)
+        return geo_prop
+
+
+@dataclass
+class LinkResource:
+    res_id: str
+    label: str
+    link_to: list[str]
+    comments: list[str]
+    permissions: str = "res-default"
+
+    def __post_init__(self) -> None:
+        _check_and_warn_strings(self.res_id, self.res_id, "Resource ID")
+        _check_and_warn_strings(self.res_id, self.label, "Label")
+
+    def serialise(self) -> etree._Element:
+        res_ele = self._serialise_resource_element()
+        res_ele.append(_serialise_has_comment(self.comments, self.res_id))
+        res_ele.append(self._serialise_links())
+        return res_ele
+
+    def _serialise_resource_element(self) -> etree._Element:
+        attribs = {"label": self.label, "id": self.res_id}
+        if self.permissions:
+            attribs["permissions"] = self.permissions
+        return etree.Element(f"{DASCH_SCHEMA}link", attrib=attribs, nsmap=XML_NAMESPACE_MAP)
+
+    def _serialise_links(self) -> etree._Element:
+        vals = [LinkValue(value=x, prop_name="hasLinkTo", resource_id=self.res_id) for x in self.link_to]
+        prop_ele = vals[0].make_prop()
```

Comment:
I don't understand what you propose? Do you want this function to take the vals as a parameter? from the order 122 cannot be above 121, because it does not exist yet...

Reply:
No, I suggest you remove the `LinkValue.make_prop()` method. It is at the wrong place there. Instead, you do this:

```
def _serialise_links(self) -> etree._Element:
        vals = [LinkValue(value=x, prop_name="hasLinkTo", resource_id=self.res_id) for x in self.link_to]
        prop_ele = etree.Element(f"{DASCH_SCHEMA}resptr-prop", name="hasLinkTo", nsmap=XML_NAMESPACE_MAP)
        prop_ele.extend([v.make_element() for v in vals])
        return prop_ele
```

Reply:
I could do that, but I prefer to reuse classes, so that changes are easier. but I can live with it.

Reply:
What do you mean with "reuse classes"?